### PR TITLE
P2 signup: add 'join workspace' step

### PR DESCRIPTION
**Pull request by annemirasol**
#### Changes proposed in this Pull Request

* Adds a new step `p2-join-workspace` to the signup flow, as part of 4326-gh-Automattic/p2
* Fixes CSS issues listed in 5136-gh-Automattic/p2

#### Testing instructions

`ENABLE_FEATURES=no-force-sympathy yarn start` on this branch.

1. Prepare the test P2 workspace. You will need to be admin to do this.
* Go to `10minutemail.com`, or similar temp mail providers. Grab the domain part of the temp email provided.
* Go to `https://wordpress.com/settings/general/YOUR-TEST-P2-WORKSPACE.wordpress.com` and add the domain as a pre-approved domain under the "Joining this workspace" section.

2. Go to `calypso.localhost:3000/start/p2` and sign up for an account using the temp email in Step 1. If on Chrome incognito, please turn off "Block third-party cookies" for testing convenience. 
3. Note: When you get redirected after clicking the confirmation email link, please make sure that you change `wordpress.com` back to `calypso.localhost:3000` in the URL.
4. After the "Complete your profile" step, you should see the new step. 
5. You should see your test P2 workspace listed as a workspace that the new user can join.
6. Verify that the code input form works as expected and you can join the workspace successfully.

Screen Shot 2022-04-20 at 6 15 19 PM

Screen Shot 2022-04-21 at 12 42 12 AM

### CODE REVIEW

File: client/assets/stylesheets/_p2-vars.scss [hunk 1]
```diff
@@ -17,6 +17,8 @@
 	--p2-color-button: #f3f3f3;
 	--p2-color-button-dark: #eaeaea;
 
+	--p2-color-background-light: #f3f3f3;
```
Comment (lamosty) [inline client/assets/stylesheets/_p2-vars.scss:20]: Seems like we already have these colors defined just above these definitions. Should we use those variables or do you think it's better if we duplicate them but with more suitable naming like you did?

File: client/signup/steps/p2-complete-profile/style.scss [hunk 2]
```diff
@@ -57,6 +57,7 @@
 	display: inline;
 	width: auto;
 	color: var( --p2-color-link );
+	font-size: inherit;
```
Comment (lamosty) [inline client/signup/steps/p2-complete-profile/style.scss:60]: 👍

File: client/signup/steps/p2-join-workspace/style.scss [hunk 3]
```diff
@@ -0,0 +1,204 @@
+.is-p-2-join-workspace .p2-step-wrapper {
+	max-width: 620px;
+
+	button.components-button.is-busy {
+		background-image: none;
+	}
+}
+
+.p2-join-workspace__workspace-list {
+	margin-bottom: 2.5em;
```
Comment (lamosty) [inline client/signup/steps/p2-join-workspace/style.scss]: I think I'm seeing the usage of `em` for margins, paddings, font-sizes, etc for the first time in Calypso codebase. Is it a new recommended practice?

File: client/signup/steps/p2-join-workspace/code-input.jsx [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+		// Reject invalid characters.
+		if ( event.key.match( CHALLENGE_CODE_ALLOWED_CHARS ) === null ) {
+			event.preventDefault();
+			return false;
+		}
+
+		return true;
+	};
+
+	const handleCodeInputKeyUp = ( event ) => {
```
Comment (lamosty) [inline client/signup/steps/p2-join-workspace/code-input.jsx:34]: Nice touch 👍

File: client/assets/stylesheets/_p2-vars.scss (hunk 1 above)
Comment (annemirasol) [reply client/assets/stylesheets/_p2-vars.scss:20]: I did that on purpose as there have been times that Edu/David tweaked the color values. Separate naming for separate elements would avoid confusion there.

**Review by lamosty — APPROVED** [review]
Tests out well and code is looking really good.

LGTM! :shipit:

File: client/signup/steps/p2-join-workspace/style.scss (hunk 3 above)
Comment (annemirasol) [reply client/signup/steps/p2-join-workspace/style.scss]: Whoops, that should have been `rem` instead. Fixed~

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7231573

Thank you @annemirasol for including a screenshot in the description! This is really helpful for our translators.

File: client/signup/steps/p2-join-workspace/code-input.jsx [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+	const renderCodeInputFieldset = () => {
+		const fieldset = [];
+		for ( let index = 0; index < CHALLENGE_CODE_LENGTH; index++ ) {
+			fieldset.push(
+				<input
+					key={ index }
+					className="p2-join-workspace__code-input-box"
+					maxLength={ 1 }
+					onChange={ ( event ) => {
+						handleCodeInputChange( event, index );
```
Comment (tyxla) [inline client/signup/steps/p2-join-workspace/code-input.jsx:130]: Seems like passing `index` here is unnecessary.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7231573).

**Pull request merged**